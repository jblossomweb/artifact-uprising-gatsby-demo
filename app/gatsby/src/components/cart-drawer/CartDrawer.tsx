import React from "react"
import Img, { FixedObject } from "gatsby-image"
import { makeStyles } from "@material-ui/core/styles"
import Box from "@material-ui/core/Box"
import Card from "@material-ui/core/Card"
import CardActions from "@material-ui/core/CardActions"
import CardContent from "@material-ui/core/CardContent"
import Button from "@material-ui/core/Button"
import Typography from "@material-ui/core/Typography"
import List from "@material-ui/core/List"
import ListItem from "@material-ui/core/ListItem"
import ListItemAvatar from "@material-ui/core/ListItemAvatar"
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction"
import ListItemText from "@material-ui/core/ListItemText"
import Avatar from "@material-ui/core/Avatar"
import TextField from "@material-ui/core/TextField"
import IconButton from "@material-ui/core/IconButton"
import Tooltip from "@material-ui/core/Tooltip"
import FolderIcon from "@material-ui/icons/Folder"
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart"
import DeleteIcon from "@material-ui/icons/Delete"
import ClearIcon from "@material-ui/icons/Clear"

import * as Style from "./CartDrawer.style"

const useStyles = makeStyles({
  root: {
    width: 400,
    overflowY: "auto",
    maxWidth: "100vw",
    boxShadow: "none",
  },
})

import { Cart, CartItem } from "../../types"

export interface Props {
  cart: Cart
  productImages?: {
    [id: number]: {
      fixed: FixedObject
    }
  }
  setQty: (id: CartItem["id"], qty: CartItem["qty"]) => void
  closeCart: () => void
}

const CartDrawer: React.FC<Props> = ({
  cart,
  productImages,
  setQty,
  closeCart,
}) => {
  const classes = useStyles()
  const isEmpty: boolean = !cart.count || !cart.items.length
  return (
    <Card className={classes.root}>
      <CardContent>
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          width="100%"
        >
          <IconButton onClick={closeCart} style={{ marginTop: -8 }}>
            <ClearIcon fontSize={"small"} />
          </IconButton>
          <Typography gutterBottom variant="h5" component="h3">
            {cart.count || 0} item{cart.count !== 1 && "s"} in your cart
          </Typography>
        </Box>
      </CardContent>
      {!isEmpty ? (
        <List dense style={{ width: "100%" }}>
          {cart.items.map(item => (
            <ListItem key={item.id}>
              <ListItemAvatar>
                <Avatar>
                  {productImages && productImages[item.id] ? (
                    <Img fixed={productImages[item.id].fixed} />
                  ) : (
                    <FolderIcon />
                  )}
                </Avatar>
              </ListItemAvatar>
              <ListItemText
                primary={item.title}
                secondaryTypographyProps={{ component: "div" }}
                secondary={
                  <Style.QtyField>
                    <TextField
                      variant="outlined"
                      type="number"
                      value={item.qty}
                      inputProps={{ min: 0, max: 99 }}
                      onChange={({ target: { value } }) => {
                        setQty(item.id, Number(value))
                      }}
                    />
                    &nbsp;x&nbsp;{item.price}&nbsp;
                    <Tooltip title="remove all">
                      <DeleteIcon
                        onClick={() => {
                          setQty(item.id, 0)
                        }}
                      />
                    </Tooltip>
                  </Style.QtyField>
                }
                style={{ maxWidth: "70%" }}
              />
              <ListItemSecondaryAction>{item.ext}</ListItemSecondaryAction>
            </ListItem>
          ))}
        </List>
      ) : null}

      <CardContent>
        <Box display="flex" justifyContent="flex-end" width="100%">
          <Typography gutterBottom variant="h5" component="h3">
            Total: {cart.total}
          </Typography>
        </Box>
      </CardContent>
      <CardActions>
        <Box display="flex" justifyContent="flex-end" width="100%">
          <Button size="large" color="primary" disabled={isEmpty}>
            Checkout&nbsp;
            <ShoppingCartIcon />
          </Button>
        </Box>
      </CardActions>
    </Card>
  )
}

export default CartDrawer
