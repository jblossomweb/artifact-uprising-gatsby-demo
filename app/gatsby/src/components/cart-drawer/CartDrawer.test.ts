import snapshotTests from "../../../utils/story-snapshot-tests"
import { stories } from "./CartDrawer.stories"

describe("components/cart-drawer", () => {
  snapshotTests(stories)
})
