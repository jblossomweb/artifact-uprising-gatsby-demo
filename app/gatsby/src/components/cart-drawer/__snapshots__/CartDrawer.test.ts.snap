// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/cart-drawer empty matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": -8,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
      <h3
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        0
         item
        s
         in your cart
      </h3>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <h3
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Total: 
        $0.00
      </h3>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Checkout 
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`components/cart-drawer sample matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginTop": -8,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h3
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        3
         item
        s
         in your cart
      </h3>
    </div>
  </div>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          style={
            Object {
              "maxWidth": "70%",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            School of Boxing
          </span>
          <div
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              className="sc-AxjAm ketZvX"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    max={99}
                    min={0}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="number"
                    value={1}
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-4"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
               x 
              $49.99
               
              <svg
                aria-describedby={null}
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                title="remove all"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        $49.99
      </div>
    </li>
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          style={
            Object {
              "maxWidth": "70%",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            Jet-Propelled Pogo Stick
          </span>
          <div
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              className="sc-AxjAm ketZvX"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    max={99}
                    min={0}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="number"
                    value={2}
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-4"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
               x 
              $29.99
               
              <svg
                aria-describedby={null}
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                title="remove all"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        $59.98
      </div>
    </li>
  </ul>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      <h3
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Total: 
        $79.98
      </h3>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-8"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Checkout 
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
